### PR TITLE
Upgrade sqlite and add carray

### PR DESCRIPTION
**Pull request by graydon**
This is a common prefix of some sqlite surgery needed by work in PRs both myself and @jonjove are working on. Sqlite gets bumped up to a version that has upsert support, and the `carray` extension gets added. Shouldn't really be noticeable outside of the forthcoming changes that depend on it, but I figured I'd split it out for landing on its own.

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v5.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [x] N/A If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -215,7 +215,7 @@ fi
 # extension, which is compiled-out of most platform sqlites. We therefore
 # always use our own bundled copy, version 3.26.0 at the time of this
 # writing.
-sqlite3_CFLAGS='-isystem $(top_srcdir)/lib/sqlite -DSQLITE_CORE'
+sqlite3_CFLAGS='-isystem $(top_srcdir)/lib/sqlite -DSQLITE_CORE -DSQLITE_OMIT_LOAD_EXTENSION=1'
```
Comment (jonjove) [inline configure.ac]: Nitpick: I think it would be better if this change (re-adding this define) was squashed into the first commit in this PR.

Comment (graydon) [reply configure.ac]: ok

Comment (graydon) [reply configure.ac]: fixed

**Comment by jonjove** [comment]
r+ f7e11f0

**Pull request merged**